### ISSUE TITLE
Pipestatus and command substitutions

### ISSUE THREAD

**Issue by runiq** [fish-shell/fish-shell#6998]
fish version 3.0.2.

I'd like to split the output of `fzf --print0` (which writes NUL-delimited output) on NUL bytes and write it into a variable. As far as I'm aware, that requires `fzf --print0 | string split0`, otherwise all output after the first NUL byte will be cut off.

However, `fzf` also returns [a meaningful exit code](https://github.com/junegunn/fzf/blob/0.21.1/src/constants.go#L84) if it was aborted, and I'd like to do something with that exit code. Piping `fzf` into another command gives me no way to get at `fzf`'s exit code.

Is there a way for me to have my cake and eat it too? :)

**Comment by krobelus** [comment]
Wait, what is the use of `--print0`? doesn't fzf split by newlines already?

Anyway you can get exit status of the individual commands in the pipe with the `$pipestatus` variable. See also https://fishshell.com/docs/current/tutorial.html?highlight=pipestatus#exit-status.

**Comment by faho** [comment]
>Anyway you can get exit status of the individual commands in the pipe with the $pipestatus variable. See also https://fishshell.com/docs/current/tutorial.html?highlight=pipestatus#exit-status.

The problem with $pipestatus here is that it doesn't work with command substitutions and `set`, so the naive

```fish
set -l res (fzf --print0 | string split0)
echo $pipestatus
```

will print *just* the status of `string split0`.

**Comment by soumya92** [comment]
Also fixed by #6820

```
~> set -l res (seq 1 10 | fzf --print0 | string split0); echo $pipestatus
(select "2")
0 0 0
```

```
~> set -l res (seq 1 10 | fzf --print0 | string split0); echo $pipestatus
(^D)
0 130 1
```
